ddl/ddl_worker.go

```diff
@@ -474,6 +474,9 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 				err = w.finishDDLJob(t, job)
 				return errors.Trace(err)
 			}
+			if runJobErr != nil {
+				txn.Discard()
+			}
 			err = w.updateDDLJob(t, job, runJobErr != nil)
```

Comment:
If `runJobErr != nil`, `updateDDLJob` will not fail as well, are we sure that is expected for all cases?

Reply:
No, `updateDDLJob ` in line 480 will still be success.

Reply:
Yes, we need to update the error to the job information.